### ISSUE TITLE
Overall content and delivery

### ISSUE THREAD

**Issue by mbgoodman** [DataLucence/dev-summer2018#5]
**Content**
- Make variable and file names CONSISTENT throughout the lessons
- Archive and crowdsource supplemental slide decks
- Push _challenge questions_ to the slide decks

**Delivery**
- Instructor monitors the class; helpers monitor individual learners
- Instructor to remember to break
- Instructor to continue when 80% green and the remaining learners have help

**Comment by mbgoodman** [comment]
**Delivery**
- Post each day's notebook to Canvas 1 hour prior to class.  Rather than posting and replacing

**Comment by XuleiQin** [comment]
1. It is better to have only one file folder path for all data. To download or use different path names always made some students get lost at the beginning of each class. This made them hard to follow the next contents of the class.
2. The length of the codes should give enough time to deliver and discuss the contents for students. Current pattern made some students just mechanically coding and without receiving the contents. Some general contents can be included in the Blank version to decrease the coding efforts for the students.
3. At the end of the class, the students can process their own data using learned functions or approaches.
4. Put more exercises for different knowledge points to interact with the students. This can improve the inclusiveness of the whole classroom, both fast coders and slow coders.
